tubular/nominal.py

```diff
@@ -853,7 +881,7 @@ def transform(self, X: pd.DataFrame) -> pd.DataFrame:
             for c in self.columns:
                 X.loc[unseen_indices[c], c] = self.unseen_levels_encoding_dict[c]
         else:
```

Comment:
I've refactored the code a bit, separating out the pre & post mapping logic. For cases where we want to handle unseen levels, we do not want to run check_mappable_rows in BaseNominalTransformer as it will error, hence specifying BaseTransformer.transform in this section. I have moved the BaseTransformer.transform to that code block to prevent it running twice for cases where unseen_level_handling=False.

Let me know if you think it needs refactoring further

Reply:
think that reads better, like that you can see where the actual mapping happens without going through if/else statements - thanks for squeezing that change in!